skimage/filters/tests/test_unsharp_mask.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+from skimage.filters import unsharp_mask
+import pytest
+
+
+@pytest.mark.parametrize("shape,multichannel",
+                         [((29,), False),
+                          ((40, 4), True),
+                          ((32, 32), False),
+                          ((29, 31, 3), True),
+                          ((13, 17, 4, 8), False)])
+@pytest.mark.parametrize("dtype", [np.uint8, np.int8,
+                                   np.uint16, np.int16,
+                                   np.uint32, np.int32,
+                                   np.uint64, np.int64,
+                                   np.float32, np.float64])
+@pytest.mark.parametrize("radius", [0, 0.1, 2.0])
+@pytest.mark.parametrize("amount", [0.0, 0.5, 2.0, -1.0])
+@pytest.mark.parametrize("offset", [-1.0, 0.0, 1.0])
+@pytest.mark.parametrize("preserve", [False, True])
+def test_unsharp_masking_output_type_and_shape(
+        radius, amount, shape, multichannel, dtype, offset, preserve):
+    array = np.random.random(shape)
+    array = ((array + offset) * 128).astype(dtype)
+    if (preserve == False) and (dtype in [np.float32, np.float64]):
+        array /= max(np.abs(array).max(), 1.0)
+    output = unsharp_mask(array, radius, amount, multichannel, preserve)
+    assert output.dtype in [np.float32, np.float64]
+    assert output.shape == shape
```

Comment:
How long does the testing take?

Reply:
A few seconds, not too fast, not too slow. The operations themselves are cheap, and the problem size is small, so it should not run for long. But there are a few variations, so it adds up.

Reply:
Ok, fine for this. :) I was worried about our utilization of the CI, but couple of seconds are well affordable.